### ISSUE TITLE
showvolume's overlay isnt working always!

### ISSUE THREAD

**Issue by phkarmur** [mpv-player/mpv#7749]
- mpv version: mpv 0.32.0-120-g3db8715-dirty
- Linux Distribution and Version: Centos 7 (3.10.0-1062.18.1.el7.x86_64)
- GPU Driver Version: 440.64.00    CUDA Version: 10.2
- Sample source: https://www.dropbox.com/s/4sz6wd6cck06t2l/HS_Recording.ts?dl=0

command: https://pastebin.com/T8QSg1Rj
working volumebars log: https://pastebin.com/ASDkixNw
not working volumebars log: https://pastebin.com/8URpi3i5

I have couple of more issues with this feed too!
1. volumebars not always not working (sometime it plays, sometime not)
2. getting pixelized output (source is clean, because i can play same video in ffplay as well as vlc)
3. sometime getting "audio sync: sync_to_video=1, offset=-3600.000000" in logs and video freezes (ie 2 and 3rd program of sample ts file)
4. if i use "--track-auto-selection=no" and try to set sid, its not working

Problem #3 can be source but other problems are not because of source. So can anyone please help to resolve these? Or please let me know if im doing anything wrong or missing any options.

Thank you.

**Comment by phkarmur** [comment]
Pixelation log: Pixelation log: https://pastebin.com/sFBAakLQ 
i have created c app for command line alternative because of player doesnt supports stream's pid selection in lavfi-complex and above code is from c app.

**Comment by wm4** [comment]
Maybe you're accidentally downscaling the video to overlay size?
Your image links don't work, but even then I wouldn't go to dropbox, too slow and annoying.

**Comment by phkarmur** [comment]
ok, let me drop images here
1. need
![showvolume](https://user-images.githubusercontent.com/16241179/82667648-e7015980-9c55-11ea-8ed2-9afe0c1c1081.png)
2. pixelation
![Pixelation](https://user-images.githubusercontent.com/16241179/82667677-f41e4880-9c55-11ea-8f43-e4124789ee38.png)
3. without volumebar
![showvolume_issue](https://user-images.githubusercontent.com/16241179/82667704-0304fb00-9c56-11ea-858f-86d7f22a20da.png)

**Comment by wm4** [comment]
That looks like decoding corruption. Maybe there's more seeking going on, or packet loss?

**Comment by phkarmur** [comment]
Im not sure if mpv discards packet because of high bitrate (60 mbps) but there isnt any packet loss in source or network.

**Comment by wm4** [comment]
Does it play OK without the lavfi stuff in mpv?

**Comment by phkarmur** [comment]
Yes, but pixelation (decoding coruption) isnt happening often. Sometime i see this issue after few hours of playing video and sometime it takes couple of days.

**Comment by wm4** [comment]
So is the "pixelation" related to the overlay stuff or not?

**Comment by phkarmur** [comment]
No, pixelation happens with and without overall.
And its pixelate only source video not overlay volumebars

**Comment by richardpl** [comment]
You do not get any overlay because you use asetrate, remove it.
asetrate changes time stamps, and is not needed.
Also you should set showvolume frame rate(rate option name) to same frame rate as your video input.

```
mpv ~/Downloads/HS_Recording.ts -lavfi-complex="[aid1]asplit[ao],showvolume=w=1030:h=40:o=1:f=0:r=10:dm=0:dmc=yellow:v=0:ds=log:b=5:p=0.5:s=1[vol1];[aid2]showvolume=w=1030:h=40:o=1:f=0:r=10:dm=0:
dmc=yellow:v=0:ds=log:b=5:p=0.5:s=1[vol2];[vid1][vol1]overlay=eval=0:x=35:y=15[v2];[v2][vol2]overlay=eval=0:x=1750:y=15[vo]"
```

**Comment by phkarmur** [comment]
I have tried without asetpts too, that not the problem.
And to about frame rate, i will try and let you know but if increase framerate, volume meter (speed) will be changing to fast.

**Comment by richardpl** [comment]
You apparently have some issues in understanding what i wrote.
Try my command, it works. Your can not work because you use ASETRATE.
I nowhere mentioned asetpts.

**Comment by phkarmur** [comment]
Sorry i understood what you wrote but i mistakenly wrote asetpts.

**Comment by phkarmur** [comment]
@richardpl if i try to use vid1, aid1 and aid2, it works always but if i change aid/vid indexes to other at that time its not working.
i have tried this command:  `mpv --msg-level=all=debug HS_Recording.ts -lavfi-complex="[aid5]showvolume=w=1030:h=40:o=1:f=0:r=10:dm=0:dmc=yellow:v=0:ds=log:b=5:p=0.5[vol1];[aid4]showvolume=w=1030:h=40:o=1:f=0:r=10:dm=0:dmc=yellow:v=0:ds=log:b=5:p=0.5[vol2];[vid4][vol1]overlay=eval=0:x=35:y=15[v2];[v2][vol2]overlay=eval=0:x=1750:y=15[vo]" --no-aid --log-file=volumebar.log`

log: https://pastebin.com/yJEy6em1
ouput: https://www.dropbox.com/s/53hbaj458j3osjg/output.ts?dl=0

**Comment by richardpl** [comment]
So you tried my proposed lavfi graph on file you uploaded?

**Comment by phkarmur** [comment]
@richardpl yes, if i try to use vid1, aid1 and aid2, it works always but if i change aid/vid indexes to other at that time sometime its not working.
i have tried this command: `mpv --msg-level=all=debug HS_Recording.ts -lavfi-complex="[aid5]showvolume=w=1030:h=40:o=1:f=0:r=10:dm=0:dmc=yellow:v=0:ds=log:b=5:p=0.5[vol1];[aid4]showvolume=w=1030:h=40:o=1:f=0:r=10:dm=0:dmc=yellow:v=0:ds=log:b=5:p=0.5[vol2];[vid4][vol1]overlay=eval=0:x=35:y=15[v2];[v2][vol2]overlay=eval=0:x=1750:y=15[vo]" --no-aid --log-file=volumebar.log`

Is it because aids and vid arent from same program/channel?

log: https://pastebin.com/yJEy6em1
ouput: https://www.dropbox.com/s/53hbaj458j3osjg/output.ts?dl=0

**Comment by richardpl** [comment]
It is because timestamps differs between two audio streams and video stream.
Use asetpts and setpts filter:

Try this demo:

```
mpv ~/Downloads/HS_Recording.ts -lavfi-complex="[aid5]asetpts=PTS-STARTPTS,showvolume=w=1030:h=40:o=1:f=0:r=10:dm=0:dmc=yellow:v=0:ds=log:b=5:p=0.5[vol1];[aid4]asetpts=PTS-STARTPTS,showvolume=w=1
030:h=40:o=1:f=0:r=10:dm=0:dmc=yellow:v=0:ds=log:b=5:p=0.5[vol2];[vid4][vol1]overlay=eval=0:x=35:y=15[v2];[v2][vol2]overlay=eval=0:x=W-w-10:y=15[vo]"
```

**Comment by phkarmur** [comment]
@richardpl thanks, its working since yesterday without any issues but its taking more cpu usage but i think i dont have any other options.

now only one problem remaining for me is pixelation/corruption `getting pixelized output (source is clean,  i can play same video in ffplay as well as vlc)` . What can be issue here? 
source: https://www.dropbox.com/s/09vtc0a9wl0c1hd/HS_Recording_2.ts?dl=0
output: https://www.dropbox.com/s/qofxhqtx3k10waa/output.ts?dl=0
